esda/lee.py

```diff
@@ -0,0 +1,143 @@
+    return result
+
+def RSpatial_Pearson(x,y,W, zero_policy=None, NAOK=False):
+    return _lee_from_R(x,y,W, local=False, zero_policy=zero_policy, NAOK=NAOK)
+
+def RLocal_Spatial_Pearson(x,y,W, zero_policy=None, NAOK=False):
+    return _lee_from_R(x,y,W, local=True, zero_policy=zero_policy, NAOK=NAOK)
+
+class Spatial_Pearson(BaseEstimator):
+    def __init__(self, connectivity=None, permutations=999):
+        self.connectivity = connectivity
+        self.permutations = permutations
+
+    def fit(self, x, y):
+        """
+        bivariate spatial pearson's R based on Eq. 18 of Lee (2001). 
+        """
+        x = utils.check_array(x)
+        y = utils.check_array(y)
+        Z = numpy.column_stack((preprocessing.StandardScaler().fit_transform(x),
+                                preprocessing.StandardScaler().fit_transform(y)))
+        if self.connectivity is None:
+            self.connectivity = sparse.eye(Z.shape[0])
+        self.association_ = self.__statistic(Z, self.connectivity) 
+
+        if (self.permutations is None):
+            return self
+        elif self.permutations < 1:
+            return self
+        
+        if self.permutations:
+            simulations = [self.__statistic(numpy.random.permutation(Z), self.connectivity)
+                           for _ in range(self.permutations)]
+            self.reference_distribution_ = simulations = numpy.array(simulations)
+            above = simulations >= self.association_
+            larger = above.sum()
+            if (self.permutations - larger) < larger:
+                larger = self.permutations - larger
+            self.significance_ = (larger + 1.) / (self.permutations + 1.)
+        return self
+
+    @staticmethod
+    def __statistic(Z,W):
+        ctc = W.T @ W
+        ones = numpy.ones(ctc.shape[0])
+        return (Z.T @ ctc @ Z) / (ones.T @ ctc @ ones)
+
+
+class Local_Spatial_Pearson(BaseEstimator):
+    def __init__(self, connectivity = None, permutations=999):
+        self.connectivity = connectivity
+        self.permutations = permutations
+
+    def fit(self, x, y):
+        """
+        bivariate local pearson's R based on Eq. 22 in Lee (2001), using 
+        site-wise conditional randomization from Moran_Local_BV.
+        """
+        x = utils.check_array(x)
+        y = utils.check_array(y)
+        #x = preprocessing.StandardScaler().fit_transform(x)
+        #y = preprocessing.StandardScaler().fit_transform(y)
+        Z = numpy.column_stack((preprocessing.StandardScaler().fit_transform(x),
+                                preprocessing.StandardScaler().fit_transform(y)))
+        
+        #slx = self.connectivity @ x
+        #sly = self.connectivity @ y
+        n, _ = x.shape
+        #self.associations_ = n * (slx - x.mean()) * (sly - y.mean())
+        #self.associations_ /= x.std() * y.std() # should always be one
+
+        self.associations_ = self.__statistic(Z,self.connectivity)
+
+        if self.permutations:
```

Comment:
Looks good for Equation (22) in Lee(2001), along with `__statistic(Z,W)` 👍 , but some inline comments would be helpful here.